src/agent/advertising_proxy.cpp

```diff
@@ -306,6 +241,101 @@ void AdvertisingProxy::PublishHostHandler(const char *aName, otbrError aError)
     }
 }
 
+void AdvertisingProxy::PublishAllHostsAndServices(void)
+{
+    const otSrpServerHost *host = nullptr;
+
+    otbrLogErr("Publish all services");
+    while ((host = otSrpServerGetNextHost(GetInstance(), host)) != nullptr)
+    {
+        PublishHostAndItsServices(host, nullptr);
+    }
+}
+
+otbrError AdvertisingProxy::PublishHostAndItsServices(const otSrpServerHost *aHost, OutstandingUpdate *aUpdate)
```

Comment:
```suggestion
otbrError AdvertisingProxy::PublishHostAndItsServices(const otSrpServerHost &aHost, OutstandingUpdate *aUpdate)
```
nit

Reply:
I couldn't do this. `otSrpServerHost` is an alias of `void` and it's not valid to use `const void &`.